crates/turborepo-lib/src/gitignore.rs

```diff
@@ -0,0 +1,141 @@
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+use std::{
+    fs::{File, OpenOptions},
+    io,
+    io::{BufRead, Write},
+};
+
+use turbopath::AbsoluteSystemPathBuf;
+
+const TURBO_GITIGNORE_ENTRY: &str = ".turbo";
+const GITIGNORE_FILE: &str = ".gitignore";
+
+pub fn ensure_turbo_is_gitignored(repo_root: &AbsoluteSystemPathBuf) -> Result<(), io::Error> {
```

Comment:
Complete nit, but we should take a reference to a borrowed varaint of the path instead of the owned one.
See [clippy lint](https://rust-lang.github.io/rust-clippy/master/#/ptr_arg)
```suggestion
pub fn ensure_turbo_is_gitignored(repo_root: &AbsoluteSystemPath) -> Result<(), io::Error> {
```